File: php/features.php
<?php
declare(strict_types=1);

// MbtaV3 SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class MbtaV3Features
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new MbtaV3BaseFeature();
            case "test":
                return new MbtaV3TestFeature();
            default:
                return new MbtaV3BaseFeature();
        }
    }
}

File: rb/features.rb
# MbtaV3 SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module MbtaV3Features
  def self.make_feature(name)
    case name
    when "base"
      MbtaV3BaseFeature.new
    when "test"
      MbtaV3TestFeature.new
    else
      MbtaV3BaseFeature.new
    end
  end
end
